File: 백준/Bronze/2581. 소수/소수.py
import math

M = int(input())
N = int(input())

sum = 0
yak_su = []
for i in range(M, N+1):
    if i >= 2:
        for j in range(2, int(math.sqrt(i)) + 1):
            if i % j == 0:
                break
        else:
            sum += i
            yak_su.append(i)

print(sum) if yak_su else 0
print(min(yak_su)) if yak_su else print(-1)

File: 백준/Bronze/2581. 소수/소수.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        Scanner sc = new Scanner(System.in);

        int M = sc.nextInt();
        int N = sc.nextInt();
        int sum = 0;
        ArrayList<Integer> yak_su = new ArrayList<>();

        for (int i = M; i < N+1; i++){
            int count = 0;
            if (i > 1){
                for (int j = 2; j < Math.floor(Math.sqrt(i)) + 1; j++){
                    if (i % j == 0){
                        count =+ 1;
                        break;
                    }

                }
                if (count == 0) {
                    yak_su.add(i);
                    sum += i;
                }
                
            }
        }
        if (yak_su.size() > 0){
            System.out.println(sum);
            System.out.println(Collections.min(yak_su));
        }
        else{
            System.out.println(-1);
        }
        
    }
}